export: Export `zerver_submessage` table.

PieterCK opened the pull request:
Fixes #36150.


**How changes were tested:**
- Imported realm:
  <img width="1104" height="511" alt="image" src="https://github.com/user-attachments/assets/c0772533-eed4-4814-80e2-8c215f55d689" />

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [ ] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [ ] Each commit is a coherent idea.
- [ ] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [ ] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

mateuszmandera commented on zerver/tests/test_import_export.py:
```diff
@@ -1518,6 +1519,34 @@ def test_import_realm(self) -> None:
+            poll_widget_message,
+            poll_message_topic,
+        )
+        export_realm_poll_submessage = SubMessage.objects.get(
```
can we name this variable and `export_realm_todo_submessage` without the `export_` prefix? that's more consistent with the naming convention in this test I think. it's the post-import objects that just get the `imported_....` naming.

PieterCK commented:
Thanks for reviewing this! I've addressed your comments, the only outstanding issue is:
- https://github.com/zulip/zulip/pull/36988#discussion_r2615173548

timabbott commented on zerver/lib/export.py:
```diff
@@ -2773,6 +2781,15 @@ def get_single_user_config() -> Config:
+        model=SubMessage,
+        normal_parent=user_profile_config,
+        include_rows="sender_id__in",
+        limit_to_consenting_users=False,
```
Correct answer is False, which is what we have. We could add a comment to clarify why it's correct.
